src/main/java/org/openstreetmap/atlas/checks/validation/tag/InvalidLanesTagCheck.java

```diff
@@ -0,0 +1,168 @@
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        if (this.isChecked.contains(object.getIdentifier()) || !partOfTollBooth(object))
+        {
+            this.markAsFlagged(object.getOsmIdentifier());
+            return Optional.of(this.createFlag(object,
+                    this.getLocalizedInstruction(0, object.getOsmIdentifier())));
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    /**
+     * Checks for a node with {@code barrier=toll_booth} amongst the {@link Edge}s that are
+     * connected to the input {@link Edge} and have an otherwise invalid {@code lanes} tag.
+     *
+     * @param object
+     *            an {@link Edge} with an otherwise invalid {@code lanes} tag, to be checked for
+     *            toll booths
+     * @return a boolean that is true when a toll booth is found
+     */
+    private boolean partOfTollBooth(final AtlasObject object)
+    {
+        final HashSet<Edge> connectedInvalidEdges = connectedInvalidLanes(object);
+
+        // check for toll booths
+        for (final Edge edge : connectedInvalidEdges)
+        {
+            for (final Node node : edge.connectedNodes())
+            {
+                if (Validators.isOfType(node, BarrierTag.class, BarrierTag.TOLL_BOOTH))
+                {
+                    // If there is a toll booth, mark them so we don't process
+                    // items twice unnecessarily, and return true
+                    connectedInvalidEdges
+                            .forEach(validEdge -> this.markAsFlagged(validEdge.getOsmIdentifier()));
+                    return true;
+                }
+            }
+        }
+        // If not a toll booth, mark for flagging so they can skip this toll booth check.
+        connectedInvalidEdges
+                .forEach(invalidEdge -> this.isChecked.add(invalidEdge.getIdentifier()));
+        return false;
+    }
+
+    /**
+     * Gets the {@link Edge}s that are connected to the input {@link Edge} and have an otherwise
+     * invalid {@code lanes} tag.
+     *
+     * @param object
+     *            an {@link Edge} with an invalid {@code lanes} tag
+     * @return a {@link HashSet} of connected invalid {@code lanes} tag {@link Edge}s
+     */
+    private HashSet<Edge> connectedInvalidLanes(final AtlasObject object)
+    {
+        // Connected edges with lanes tag values not in the lanesFilter
+        final HashSet<Edge> connectedEdges = new HashSet<>();
+        // Queue of edges to be processed
+        final ArrayDeque<Edge> toProcess = new ArrayDeque<>();
+        Edge polledEdge;
+
+        // Add original edge
+        connectedEdges.add((Edge) object);
+        toProcess.add((Edge) object);
+
+        // Get all connected edges with lanes tag values not in the lanesFilter
+        while (!toProcess.isEmpty())
```

Comment:
You might want to also want to have a size limit on the connected edges that checked for here. The if statement below does limit the likelihood of this getting out of control, but under a very specific set of circumstances you could get into a situation where this loops for quite a while.